aten/src/ATen/native/UpSample.h

```diff
@@ -162,12 +162,17 @@ template <typename scalar_t>
 static inline scalar_t linear_upsample_compute_source_index(
     scalar_t scale,
     int64_t dst_index,
-    bool align_corners) {
+    bool align_corners,
+    bool cubic) {
   if (align_corners) {
     return scale * dst_index;
   } else {
     scalar_t src_idx = scale * (dst_index + 0.5) - 0.5;
-    return src_idx < 0 ? scalar_t(0) : src_idx;
+    // [Note] Follow Opencv resize logic:
+    // When doing cubic interpolation, we allow negative src_idx
+    //   to compute dx = src_idx - floorf(src_idx) later.
+    // For other modes, it's bounded by 0.
+    return (!cubic && src_idx < 0) ? scalar_t(0) : src_idx;
```

Comment:
Really this could be its own function instead of a bool flag, it's kind of weird anyways calling something named `linear_upsample_compute_source_index` inside a cubic kernel

Reply:
Actually the name is incorrect, it's not specific to linear. It's only related to how we compute distance (like using area mode or not). 
I would prefer to keep this bool flag and rename the function so that this logic is in one place.

Reply:
as @ppwwyyxx suggested it might be save to just use negative index to interpolate for linear modes, so we can indeed remove the flag. Let's see if the tests are happy or not :D

Reply:
hmmm tests failed due to the way we wrote linear interpolations didn't bound indices. This would require some changes to get it fixed. I will make it a todo and keep the bool flag.